### PR TITLE
Simplify append offset of legend

### PR DESCRIPTION
**Pull request by ckifer**
## Description

Remove DOM access one thing at a time.

_technically_ this doesn't remove DOM access, but it does remove an unnecessary call to getChildByType and a lot of other stuff that isn't really being used for anything. Yay!

Also we don't need to worry about compat anymore in 3.x so remove export and remove `getLegendProps`

## Motivation and Context

Above. Remove some code.

## How Has This Been Tested?

npm test

Wait for chromatic diff

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Refactor

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] I have added a storybook story or extended an existing story to show my changes

### CODE REVIEW

File: src/chart/generateCategoricalChart.tsx [hunk 1]
```diff
@@ -736,8 +736,7 @@ const calculateOffset = (
   }
 
   if (legendItem && prevLegendBBox) {
-    // @ts-expect-error margin is optional in props but required in appendOffsetOfLegend
-    offset = appendOffsetOfLegend(offset, props, prevLegendBBox);
+    offset = appendOffsetOfLegend(offset, legendItem, prevLegendBBox);
```
Comment (ckifer) [inline src/chart/generateCategoricalChart.tsx:739]: we only passed in props to get `legendItem` again using `findChildByType`, but we quite literally already have it above. Why find it twice? Don't!

File: src/util/getLegendProps.ts
Comment (ckifer) [inline src/util/getLegendProps.ts:1]: none of the props that are created from merging props and the result of `getWidthOrHeight` are actually used. So we really don't need this

**Comment by ckifer** [comment]
No diff https://www.chromatic.com/build?appId=63da8268a0da9970db6992aa&number=1112

**Comment by ckifer** [comment]
Codecov failing because lines were removed

**Review by PavelVanecek — APPROVED** [review]
Nice!

**Pull request merged**